qiskit/algorithms/gradients/base_qfi.py

```diff
@@ -0,0 +1,212 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Abstract base class of the Quantum Fisher Information (QFI) for ``Estimator``.
+"""
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from collections.abc import Sequence
+from copy import copy
+
+from qiskit.circuit import Parameter, QuantumCircuit
+from qiskit.opflow import PauliSumOp
+from qiskit.primitives import BaseEstimator
+from qiskit.providers import Options
+from qiskit.algorithms import AlgorithmJob
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+from .qfi_result import QFIResult
+
+
+class BaseQFI(ABC):
```

Comment:
In looking at the comment above `Abstract base class of the Quantum Fisher Information (QFI) for ``Estimator``.` would we ever have a sampler version? I note we have BaseEstimatorGradient, with Estimator in the name, but estimator is not part of this class name, i.e its BaseQFI not BaseEstimatorQFI. If there was a possibility to add Sampler variant later then might this make sense?

Reply:
There shouldn't be a sampler/estimator QFI since QFI is a property of the parametrized circuit only.

Reply:
I think its more than just the text since it has estimator as part of the constructor - Stefan said he will touch base with you as I asked him offline to elaborate on his comment.

Reply:
I would remove `Estimator` from the base class.